pkg/database/database_env.go

```diff
@@ -0,0 +1,55 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+const (
+	dbConnectionErrStr string = "Could not create database connection string: %s must be set"
+
+	DbKindEnvVar     string = "DATABASE_KIND"
+	DbNameEnvVar     string = "DATABASE_DB"
+	DbUserEnvVar     string = "DATABASE_USER"
+	DbPasswordEnvVar string = "DATABASE_PASSWORD" // #nosec G101 not a password
+	DbHostEnvVar     string = "DATABASE_URL"
+	DbPortEnvVar     string = "DATABASE_PORT"
+)
+
+type databaseEnvironment struct {
+	kind     string
+	name     string
+	user     string
+	password string
+	host     string
+	port     string
+}
+
+func newDatabaseEnvironment() (*databaseEnvironment, error) {
+	var err error
+
+	kind := getEnv(DbKindEnvVar, err)
+	name := getEnv(DbNameEnvVar, err)
+	user := getEnv(DbUserEnvVar, err)
+	password := getEnv(DbPasswordEnvVar, err)
+	host := getEnv(DbHostEnvVar, err)
+	port := getEnv(DbPortEnvVar, err)
```

Comment:
Reeturn the error instead:

```suggestion
	kind, err := getEnv(DbKindEnvVar, err)
	name, err := getEnv(DbNameEnvVar, err)
	user, err := getEnv(DbUserEnvVar, err)
	password, err := getEnv(DbPasswordEnvVar, err)
	host, err := getEnv(DbHostEnvVar, err)
	port, err := getEnv(DbPortEnvVar, err)
```

And inside the `getEnv` join if there is an error and return.

Because currently this does not work, and we shouldn't pass a pointer to an error. At least its now working for me on this example:

```
package main

import (
	"errors"
	"fmt"
)

func a(err error) {
	err = errors.Join(err, fmt.Errorf("Something bad happened"))
}

func main() {
	var err error
	a(err)

	if err != nil {
		fmt.Println("There was an error")
	}

}
```

Reply:
ops, you are right, I missed this

Reply:
I fixed it by passing the pointer (my intention at the beginning but I missed it)

Reply:
it's less explicit, in general I wouldn't do it that way but it's a helper private method and it's point is to save some lines of code, and for this situation I think could be ok

Reply:
Nevermind, I had something wrong.